new-log-viewer/src/services/MainWorker.worker.ts

```diff
@@ -0,0 +1,39 @@
+import {
+    MainWorkerReqMessage,
+    WORKER_PROTOCOL_REQ,
+    WORKER_PROTOCOL_RESP,
+} from "../typings/worker";
+
+
+onmessage = (ev: MessageEvent<MainWorkerReqMessage>) => {
+    const {code, args} = ev.data;
+    console.log(`[Render -> MainWorker] code=${code}: args=${JSON.stringify(args)}`);
+
+    switch (code) {
+        case WORKER_PROTOCOL_REQ.LOAD_FILE: {
+            postMessage({
+                code: WORKER_PROTOCOL_RESP.PAGE_DATA,
+                args: {
+                    logs: "Hello world!",
+                    lines: [1],
```

Comment:
Sorry, I was using an older diagram when I came up with this example. Yes, it should be a `Map<number, number>` with the key being (first) line number and the value being log event number.
By the way, for consistency, we name indices that start with 1 as "num" / "number" and those who start with 0 as "idx" / "index". Let me know if that would cause any confusion.

Reply:
Yeah, I figured. Probably worth adding a comment somewhere.